pkg/server/server.go

```diff
@@ -633,7 +633,7 @@ func (s *applicationServer) AuthorizeGitlab(ctx context.Context, msg *pb.Authori
 		return nil, fmt.Errorf("could not exchange code: %w", err)
 	}
 
-	token, err := s.jwtClient.GenerateJWT(time.Duration(tokenState.ExpiresIn), gitproviders.GitProviderGitLab, tokenState.AccessToken)
+	token, err := s.jwtClient.GenerateJWT(time.Duration(tokenState.ExpiresIn)*time.Second, gitproviders.GitProviderGitLab, tokenState.AccessToken)
```

Comment:
Nit but multiplying by seconds here makes the code a bit obscure.  From looking at the code it appears that only GitLab (today) uses the ExpiresIn field.  However, others might.  The code is more readable (to me) if we add a function on tokenState to return the expires value in seconds - or another field ExpiresInSeconds. Then the individual providers can provide the conversion - if necessary.

Reply:
You don't need the type assertion `time.Duration` either, because multiplying an integer by time.Duration will get you a time.Duration.